components/event-processor/org.wso2.carbon.event.processor.core/src/main/java/org/wso2/carbon/event/processor/core/EventProcessorService.java

```diff
@@ -181,7 +181,7 @@ public void setStatisticsEnabled(String executionPlanName, boolean isEnabled,
      * @param executionPlan execution plan.
      * @return  true if valid.
      */
-    public boolean validateExecutionPlan(String executionPlan);
+    public String validateExecutionPlan(String executionPlan);
```

Comment:
When a boolean is returned, 
- when the plan is invalid, the UI-client receives the boolean 'false'. The UI-client does not get any specific error message.

To overcome the above limitation, we can return the error message, extracted from the exception,  and send to the UI client.